apps/namadillo/src/registry/index.ts

```diff
@@ -0,0 +1,47 @@
+import cosmoshub from "./cosmoshub.json";
+import namada from "./namada.json";
+
+type MinimalDenom = string;
+type ConfigName = string;
+
+type Currency = {
+  coinDecimals: number;
+  coinDenom: string;
+  coinMinimalDenom: string;
+};
+
+export type ChainConfig = {
+  currencies: Currency[];
+  feeCurrencies: Currency[];
+  stakeCurrency: Currency;
+};
+
+const loadedConfigs: ChainConfig[] = [];
+const minimalDenomMap: Map<string, number> = new Map();
+const nameMap: Map<string, number> = new Map();
+
+export function chainConfigByMinDenom(minDenom: string): ChainConfig {
+  // We assume that the map always has the key
+  const index = minimalDenomMap.get(minDenom)!;
+  return loadedConfigs[index];
+}
+
+export function chainConfigByName(name: string): ChainConfig {
```

Comment:
I think we can define type like:
`type ChainName = 'cosmoshub' | 'namada';`
At this point I'm not sure though if we will always have every json loaded so it might be undefined anyways.
WDYT?

Reply:
I think this works for now :)